specification/network/resource-manager/Microsoft.Network/2017-09-01/applicationSecurityGroup.json

```diff
@@ -0,0 +1,370 @@
+              "$ref": "./network.json#/definitions/ResourceReferenceResultList"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List all IP configurations that reference application security group": { "$ref": "./examples/ApplicationSecurityGroupListIpConfigurations.json" }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/applicationSecurityGroups/{applicationSecurityGroupName}/referencingSecurityRules/$ref": {
+      "get": {
+        "tags": [
+          "ApplicationSecurityGroups"
+        ],
+        "operationId": "ApplicationSecurityGroups_ListReferencingSecurityRules",
+        "description": "This operation returns the security rules that reference the specified application security group resource.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "applicationSecurityGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the application security group."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns the security rules that reference the specified application security group resource.",
+            "schema": {
+              "$ref": "./network.json#/definitions/ResourceReferenceResultList"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List all security rules that reference application security group": { "$ref": "./examples/ApplicationSecurityGroupListReferencingSecurityRules.json" }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ApplicationSecurityGroup": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ApplicationSecurityGroupPropertiesFormat",
+          "description": "Properties of the application security group."
+        },
+        "etag": {
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/Resource"
+        }
+      ],
+      "description": "An application security group in a resource group."
+    },
+    "ApplicationSecurityGroupPropertiesFormat": {
+      "properties": {
+        "resourceGuid": {
```

Comment:
I know this property is present in every NRP swagger. How is this being used? Why does the service need it. And it most likely is read-only as the service might be generating this. But what is not clear is that how will customer use this property.

Reply:
It's a read-only property. It uniquely identifies a resource, even if the user changes its name or migrate the resource across subscriptions or resource groups.

Reply:
@murligor - You just provided information that is  pure gold.

> It uniquely identifies a resource, even if the user changes its name or migrate the resource across subscriptions or resource groups.
Can you add the above sentence to the description of resourceGuid?  It gives valuable info to the customer.

Reply:
Done